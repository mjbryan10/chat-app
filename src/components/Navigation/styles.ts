import styled from 'styled-components';

export const Nav = styled.nav`
  display: flex;
  justify-content: space-between;
  min-height: 50px;
  margin: 5px;
  h1 {
    margin: 0;
  }
`;