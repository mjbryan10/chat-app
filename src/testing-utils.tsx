import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }