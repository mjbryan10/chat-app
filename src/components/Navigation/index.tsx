import React from 'react';
import * as S from './styles';
import ThemeToggle from 'features/theme/ThemeToggle';
import AppTitle from 'components/AppTitle';
import NavigationConversationsButton from 'features/navigation/NavigationConversationsButton';
import { selectNavigationState } from 'features/navigation/navigationSlice';
import LogoutButton from 'features/login/LogoutButton';
import { useSelector } from 'react-redux';

const Navigation = () => {
   const navigationState = useSelector(selectNavigationState);
   return (
      <S.Nav>
         {navigationState !== 'index' ? (
            <NavigationConversationsButton />
         ) : (
            <LogoutButton />
         )}
         <AppTitle />
         <ThemeToggle />
      </S.Nav>
   );
};

export default Navigation;
