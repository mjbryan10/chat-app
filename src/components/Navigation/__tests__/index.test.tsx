import React from 'react';
import { render, cleanup } from 'testing-utils';
import { useSelector } from 'react-redux';
import Navigation from '..';

jest.mock('react-redux', () => ({
   ...jest.requireActual<any>('react-redux'),
   useSelector: jest.fn(),
}));

describe('Navigation Component', () => {
   afterEach(() => cleanup());

   test('should display logout button if on page index', () => {
      (useSelector as jest.Mock).mockImplementationOnce(() => 'index');
      const { getByTestId } = render(<Navigation />);

      expect(getByTestId('navigation-button-logout')).toBeVisible();
   });

   test('should display back button if not on page index', () => {
      (useSelector as jest.Mock).mockImplementationOnce(() => 'conversation');
      const { getByTestId } = render(<Navigation />);

      expect(getByTestId('navigation-button-back')).toBeVisible();
   });

   test('should render correctly', () => {
      const { asFragment } = render(<Navigation />);

      expect(asFragment()).toMatchSnapshot();
   });
});
