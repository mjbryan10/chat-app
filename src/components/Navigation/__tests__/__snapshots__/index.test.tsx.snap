// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="sc-AxjAm bjWIXH"
  >
    <button
      class="sc-AxheI cpEQhe"
      data-testid="button-navbar-icon"
    >
      <svg
        data-testid="navigation-button-back"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
        />
      </svg>
    </button>
    <h1
      class="sc-Axmtr RycXh"
    >
      Welcome to Ch@
    </h1>
    <div
      class="sc-AxhUy eRBRnr"
    >
      <button
        class="sc-AxheI cpEQhe"
        data-testid="button-navbar-icon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M195 125c0-26.3 5.3-51.3 14.9-74.1C118.7 73 51 155.1 51 253c0 114.8 93.2 208 208 208 97.9 0 180-67.7 202.1-158.9-22.8 9.6-47.9 14.9-74.1 14.9-106 0-192-86-192-192z"
          />
        </svg>
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
